py/torch_migraphx/fx/converters/acc_ops_converters.py

```diff
@@ -112,6 +113,74 @@ def acc_ops_linear(mgx_module, node, args, kwargs):
     return MGXInstruction(out_mgx)
 
 
+# NLL (negative log likelihood loss) converter.  This op assumes data has already been normalized with log_softmax
+# to give meaningful results, although there is no restriction on input values.
+@migraphx_converter(acc_ops.nll_loss)
+def acc_ops_nll_loss(mgx_module, node, args, kwargs):
+
+    inp = kwargs['input']
+    inp_instr_ref = inp.instr_ref
+    target = kwargs['target']
+    target_ref = target.instr_ref
+
+    ndims = len(inp_instr_ref.shape().lens())
+    dtype = get_arg_dtype(inp_instr_ref)
+    # weight should be a vector of 1's if not given
+    weight = mgx_module.add_literal(torch.tensor((1), dtype=dtype).numpy()) if kwargs.get('weight') is None else kwargs['weight'].instr_ref
+
+    # negative of input
+    neg_ins = mgx_module.add_instruction(migraphx.op('neg'), [inp_instr_ref])
+    #
+    # match rank of target to the input: unsqueeze
+    target_ref_unsquoze =  mgx_module.add_instruction(
+        migraphx.op('unsqueeze', axes=list(range(1, ndims))), [target_ref])
+
+    # Prepare to select weight for each target
+    # unsqueeze the weight and broadcast to match input shape 
+    #  TODO: this does not work for ndims > 2 (not currently supported)
+    weight_unsquoze = mgx_module.add_instruction(
+            migraphx.op('unsqueeze', axes=list(range(ndims-1))), [weight])
+    weight_bcst = mgx_module.add_instruction(
+            migraphx.op('multibroadcast', out_lens=inp_instr_ref.shape().lens()), [weight_unsquoze])
+
+    # Use target indexes to select weights, one for each class
+    class_axis = 1
+    weight_to_use = gather_elements(mgx_module, class_axis, [weight_bcst, target_ref_unsquoze])
+
+    # Use target indexes to select inputs
+    x_to_use = gather_elements(mgx_module, class_axis, [neg_ins, target_ref_unsquoze])
```

Comment:
```suggestion
    target_unsq = mgx_module.add_instruction(migraphx.op('unsqueeze', axes=[-1]), [target_ref])
    batch_dim_indices = mgx_module.add_literal(torch.arange(target_ref.shape().elements()).numpy())
    batch_dim_indices_unsq = mgx_module.add_instruction(migraphx.op('unsqueeze', axes=[-1]), [batch_dim_indices])
    gathernd_indices = mgx_module.add_instruction(migraphx.op('concat', axis=-1), [batch_dim_indices_unsq, target_unsq])
    
    weight_to_use = mgx_module.add_instruction(migraphx.op('gathernd'), [weight_bcst, gathernd_indices])
    x_to_use = mgx_module.add_instruction(migraphx.op('gathernd'), [neg_ins, gathernd_indices])
```
You can use the `gathernd` op here instead of the complicated stuff going on in the util functions. This should work, you'll just have to fix the reduction axis on the reduction ops below.

Reply:
I need to understand how this works better to know if it's generalizable to k dims.   It looks like it's specific to certain ranks of inputs, 1 or 2 or both?

I believe the Migraphx `unsqueeze` op can support more than one input axis, but only one negative axis input.  Not sure if that would come into play here.

Reply:
actually, this might work as it is for k dims. The `gathernd_indices` shouldnt need to change for the k dims case (thats what the unsqueeze is used for)

Reply:
only thing that might need to be modified is the output shape in the case of no reduction (although I think that should also be correct as written)

Reply:
> actually, this might work as it is for k dims. The `gathernd_indices` shouldnt need to change for the k dims case (thats what the unsqueeze is used for)

a little testing shows that `batch_dim_indices` above doesn't work as a 1-d index and would have to be created from the dimension 0 range instead, and then reshaped with more dimensions added in a way that GatherND could read the result correctly.  I think it could eventually be made to work but it's starting to look nearly as complicated as the `GatherElements` version.  So this change doesn't save us much.

Reply:
Well taking another look, it doesnt just work as it is because the target tensor is also extended by kdims. You can generalize it by treating all the N + k-dims as batch dims (apply transpose to the input so that C is the last dim) and creating a k-dimensional base coordinate grid for it rather than the 1-d I showed originally.

Reply:
I messed around a bit in my local environment, and this seems to work (might need ironing out but this is the concept):

```
target_unsq = mgx_module.add_instruction(migraphx.op('unsqueeze', axes=[-1]), [target_ref])
    target_rank = target_unsq.shape().ndim()
    # Create base coordinate tensors
    coordinate_index_lits = []
    for ax in range(target_rank-1):
        batch_dim_indices = mgx_module.add_literal(torch.arange(target_unsq.shape().lens()[ax]).numpy())
        unsq_dims = list(range(target_rank))
        unsq_dims = unsq_dims[:ax] + unsq_dims[ax+1:]
        
        batch_dim_indices_unsq = mgx_module.add_instruction(migraphx.op('unsqueeze', axes=unsq_dims), [batch_dim_indices])
        batch_dim_indices_bc = mgx_module.add_instruction(
            migraphx.op('multibroadcast', out_lens=target_unsq.shape().lens()), [batch_dim_indices_unsq])
        coordinate_index_lits.append(batch_dim_indices_bc)

    # Add the target indices to the base coordinates
    coordinate_index_lits.append(target_unsq)
    gathernd_indices = mgx_module.add_instruction(migraphx.op('concat', axis=-1), coordinate_index_lits)
    
    perm = [0] + list(range(2, ndims)) + [1]
    weight_bcst_t = mgx_module.add_instruction(migraphx.op('transpose', permutation=perm), [weight_bcst])
    neg_ins_t = mgx_module.add_instruction(migraphx.op('transpose', permutation=perm), [neg_ins])
    
    weight_to_use = mgx_module.add_instruction(migraphx.op('gathernd'), [weight_bcst_t, gathernd_indices])
    x_to_use = mgx_module.add_instruction(migraphx.op('gathernd'), [neg_ins_t, gathernd_indices])
```

Reply:
Let's address this later as out of scope.